[@remix-run/cloudflare-pages] `ERESOLVE overriding peer dependency` when creating a new Remix app

### What version of Remix are you using?

1.3.3

### Steps to Reproduce

just creating new apps and using Cloudflare pages for deployment target

```
npx create-remix@latest
? Where would you like to create your app? ./my-remix-app
? What type of app do you want to create? Just the basics
? Where do you want to deploy? Choose Remix if you're unsure, it's easy to change deployment targets. Cloudflare Pages
? TypeScript or JavaScript? TypeScript
? Do you want me to run `npm install`? Yes
```

### Expected Behavior

it should not show warn overriding peer dependency

### Actual Behavior

<details>
<summary><code>npx create-remix@latest</code></summary>
<pre>
? Where would you like to create your app? ./my-remix-app
? What type of app do you want to create? Just the basics
? Where do you want to deploy? Choose Remix if you're unsure, it's easy to change deployment targets. Cloudflare Pages
? TypeScript or JavaScript? TypeScript
? Do you want me to run `npm install`? Yes
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @remix-run/cloudflare@1.3.3
npm WARN Found: @cloudflare/workers-types@3.4.0
npm WARN node_modules/@cloudflare/workers-types
npm WARN   dev @cloudflare/workers-types@"^3.2.0" from the root project
npm WARN   1 more (@remix-run/cloudflare-pages)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer @cloudflare/workers-types@"^2.2.2" from @remix-run/cloudflare@1.3.3
npm WARN node_modules/@remix-run/cloudflare
npm WARN   @remix-run/cloudflare@"1.3.3" from @remix-run/cloudflare-pages@1.3.3
npm WARN   node_modules/@remix-run/cloudflare-pages
npm WARN
npm WARN Conflicting peer dependency: @cloudflare/workers-types@2.2.2
npm WARN node_modules/@cloudflare/workers-types
npm WARN   peer @cloudflare/workers-types@"^2.2.2" from @remix-run/cloudflare@1.3.3
npm WARN   node_modules/@remix-run/cloudflare
npm WARN     @remix-run/cloudflare@"1.3.3" from @remix-run/cloudflare-pages@1.3.3
npm WARN     node_modules/@remix-run/cloudflare-pages
npm WARN deprecated rollup-plugin-inject@3.0.2: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-inject.
</pre>
</details>

when changing dependencies (e.g. exact the version) and then running `npm install`, it's throwing an error:

```
npm install

> postinstall
> remix setup cloudflare

Error: Cannot find module '@remix-run/cloudflare/package.json'
Require stack:
- /Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/setup.js
- /Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/commands.js
- /Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/run.js
- /Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/index.js
- /Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/index.js
- /Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.resolve (node:internal/modules/cjs/helpers:108:19)
    at resolvePackageJsonFile (/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/setup.js:111:18)
    at Object.setupRemix (/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/setup.js:65:29)
    at Object.setup (/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/commands.js:132:17)
    at Object.run (/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/run.js:360:22)
    at Object.<anonymous> (/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli.js:16:11)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/setup.js',
    '/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/commands.js',
    '/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/run.js',
    '/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/index.js',
    '/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/index.js',
    '/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli.js'
  ]
}
npm ERR! code 1
npm ERR! path /Users/iamyuu/dev/sandbox/my-remix-app
npm ERR! command failed
npm ERR! command sh -c remix setup cloudflare
```

--- a/packages/remix-cloudflare-pages/package.json
+++ b/packages/remix-cloudflare-pages/package.json
@@ -21,7 +21,7 @@
     "@cloudflare/workers-types": "^3.2.0"
   },
   "devDependencies": {
-    "@cloudflare/workers-types": "^3.2.0",
+    "@cloudflare/workers-types": "^3.4.0",
     "@types/mime": "^2.0.3"
   }
 }

--- a/packages/remix-cloudflare-pages/package.json
+++ b/packages/remix-cloudflare-pages/package.json
@@ -18,7 +18,7 @@
     "@remix-run/server-runtime": "1.3.4"
   },
   "peerDependencies": {
-    "@cloudflare/workers-types": "^3.2.0"
+    "@cloudflare/workers-types": "^3.0.0"
   },
   "devDependencies": {
     "@cloudflare/workers-types": "^3.4.0",

--- a/packages/remix-cloudflare-pages/tsconfig.json
+++ b/packages/remix-cloudflare-pages/tsconfig.json
@@ -1,7 +1,7 @@
 {
   "include": ["**/*.ts"],
   "compilerOptions": {
-    "lib": ["ES2019", "WebWorker"],
+    "lib": ["ES2019"],
     "target": "ES2019",
     "types": ["@cloudflare/workers-types"],
 

--- a/packages/remix-cloudflare-workers/package.json
+++ b/packages/remix-cloudflare-workers/package.json
@@ -18,9 +18,9 @@
     "@remix-run/cloudflare": "1.3.4"
   },
   "peerDependencies": {
-    "@cloudflare/workers-types": "^2.2.2"
+    "@cloudflare/workers-types": "^2.2.2 || ^3.0.0"
   },
   "devDependencies": {
-    "@cloudflare/workers-types": "^2.2.2"
+    "@cloudflare/workers-types": "^3.4.0"
   }
 }

--- a/packages/remix-cloudflare-workers/package.json
+++ b/packages/remix-cloudflare-workers/package.json
@@ -18,7 +18,7 @@
     "@remix-run/cloudflare": "1.3.4"
   },
   "peerDependencies": {
-    "@cloudflare/workers-types": "^2.2.2 || ^3.0.0"
+    "@cloudflare/workers-types": "^2.0.0 || ^3.0.0"
   },
   "devDependencies": {
     "@cloudflare/workers-types": "^3.4.0"

--- a/packages/remix-cloudflare-workers/tsconfig.json
+++ b/packages/remix-cloudflare-workers/tsconfig.json
@@ -1,7 +1,7 @@
 {
   "include": ["**/*.ts"],
   "compilerOptions": {
-    "lib": ["ES2019", "WebWorker"],
+    "lib": ["ES2019"],
     "target": "ES2019",
     "types": ["@cloudflare/workers-types"],
 

--- a/packages/remix-cloudflare/package.json
+++ b/packages/remix-cloudflare/package.json
@@ -16,7 +16,7 @@
     "@remix-run/server-runtime": "1.3.4"
   },
   "peerDependencies": {
-    "@cloudflare/workers-types": "^2.2.2 || ^3.0.0"
+    "@cloudflare/workers-types": "^2.0.0 || ^3.0.0"
   },
   "devDependencies": {
     "@cloudflare/workers-types": "^3.4.0"

--- a/packages/remix-cloudflare/package.json
+++ b/packages/remix-cloudflare/package.json
@@ -16,9 +16,9 @@
     "@remix-run/server-runtime": "1.3.4"
   },
   "peerDependencies": {
-    "@cloudflare/workers-types": "^2.2.2"
+    "@cloudflare/workers-types": "^2.2.2 || ^3.0.0"
   },
   "devDependencies": {
-    "@cloudflare/workers-types": "^2.2.2"
+    "@cloudflare/workers-types": "^3.4.0"
   }
 }

--- a/packages/remix-cloudflare/tsconfig.json
+++ b/packages/remix-cloudflare/tsconfig.json
@@ -1,7 +1,7 @@
 {
   "include": ["**/*.ts"],
   "compilerOptions": {
-    "lib": ["ES2019", "WebWorker"],
+    "lib": ["ES2019"],
     "target": "ES2019",
     "types": ["@cloudflare/workers-types"],
 

--- a/templates/cloudflare-pages/package.json
+++ b/templates/cloudflare-pages/package.json
@@ -21,7 +21,7 @@
     "remix": "*"
   },
   "devDependencies": {
-    "@cloudflare/workers-types": "^3.2.0",
+    "@cloudflare/workers-types": "^3.4.0",
     "@remix-run/dev": "*",
     "@remix-run/eslint-config": "*",
     "@types/react": "^17.0.24",

--- a/templates/cloudflare-workers/package.json
+++ b/templates/cloudflare-workers/package.json
@@ -23,7 +23,7 @@
     "remix": "*"
   },
   "devDependencies": {
-    "@cloudflare/workers-types": "^2.2.2",
+    "@cloudflare/workers-types": "^3.4.0",
     "@remix-run/dev": "*",
     "@remix-run/eslint-config": "*",
     "@types/react": "^17.0.24",

